### PR TITLE
feat: change "incidence matrix" naming to "biadjacency matrix"

### LINKED ISSUE #900: "incidence matrix" naming should be changed to "biadjacency matrix"
Mirroring https://github.com/igraph/igraph/issues/2326, any function names referring to an "incidence matrix" should be changed to say "biadjacency matrix".

This affects at least `as_incidence_matrix()` and `graph_from_incidence_matrix()`.

In the future, the name "incidence" matrix will be used for edge-vertex incidence matrices.

@krlmlr @maelle Up to you when you start adding aliases in preparation for a future deprecation (and eventual repurposing) of the "incidence" names.

### PR DESCRIPTION
**Pull request by maelle**
Fix  #900

### CODE REVIEW

**Comment by maelle** [comment]
This becomes a small blocker for #716 since it edits the zzz-deprecate.R script.

File: R/conversion.R [hunk 1]
```diff
@@ -858,7 +858,20 @@ as_incidence_matrix <- function(graph, types = NULL, attr = NULL,
...
+#' @description
+#' `r lifecycle::badge("deprecated")`
+#'
+#' `as_incidence_matrix()` was renamed to `as_biadjacency_matrix()` to create a more
+#' consistent API.
+#' @inheritParams as_biadjacency_matrix
+#' @keywords internal
+#' @export
+as_incidence_matrix <- function(...) {
+   lifecycle::deprecate_soft("1.5.2", "as_incidence_matrix()", "as_biadjacency_matrix()")
```
Comment (maelle) [inline R/conversion.R:872]: @krlmlr what version number should we use here?

**Review by krlmlr — APPROVED** [review]

**Review by szhorvat — CHANGES_REQUESTED** [review]
Can you please update the documentation text to refer to "bipartite adjacency matrix" instead of "incidence matrix" in all places? Add just one note along these lines:

> Some authors refer to the bipartite adjacency matrix as the "bipartite incidence matrix". igraph 1.6.0 and later does not use this naming to avoid confusion with the edge-vertex incidence matrix.

**Comment by krlmlr** [comment]
@szhorvat: Please remove the "blocked" label when good.

**Comment by szhorvat** [comment]
@krlmlr Looks good to me, but are you going to do 1.5.2 instead of 1.6.0? There are a lot of changes here, worth a version bump ...

**Comment by szhorvat** [comment]
The version numbers in this PR are all 1.5.2, and @maelle was asking about about what version number you were going to use.

**Review by szhorvat — APPROVED** [review]

**Comment by szhorvat** [comment]
@krlmlr This looks fine for me, other than the version. I'll leave it to you to deal with what version is mentioned in the docs.

**Comment by krlmlr** [comment]
Thanks! This PR doesn't seem to be breaking anything.

**Comment by krlmlr** [comment]
@maelle: Can you please work on the test coverage? Probably it's sufficient to test that the deprecated variants give the same results as the original, plus a deprecation warning?

File: src/cpp11.cpp [hunk 2]
```diff
@@ -15,381 +15,381 @@ extern "C" SEXP _igraph_igraph_hcass2(SEXP n, SEXP ia, SEXP ib) {
...
-extern SEXP R_igraph_write_graph_leda(void *, void *, void *, void *);
-extern SEXP R_igraph_write_graph_lgl(void *, void *, void *, void *, void *);
-extern SEXP R_igraph_write_graph_ncol(void *, void *, void *, void *);
-extern SEXP R_igraph_write_graph_pajek(void *, void *);
-extern SEXP UUID_gen(void *);
-extern SEXP make_lazy(void *, void *, void *);
-extern SEXP make_lazy_dots(void *, void *);
-extern SEXP promise_env_(void *);
-extern SEXP promise_expr_(void *);
+extern SEXP R_igraph_add_edges(SEXP, SEXP);
```
Comment (krlmlr) [inline src/cpp11.cpp:18]: Can you please revert changes to this file?

**Review by krlmlr — APPROVED** [review]
Thanks, looks good! Can you please set the "mergequeue" label when done?

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**